sdk/core/core-rest-pipeline/src/policies/auxiliaryAuthenticationHeaderPolicy.ts

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { GetTokenOptions, TokenCredential } from "@azure/core-auth";
+import { AzureLogger } from "@azure/logger";
+import { PipelineRequest, PipelineResponse, SendRequest } from "../interfaces";
+import { PipelinePolicy } from "../pipeline";
+import { AccessTokenGetter, createTokenCycler } from "../util/tokenCycler";
+import { logger as coreLogger } from "../log";
+import { AuthorizeRequestOptions } from "./bearerTokenAuthenticationPolicy";
+
+/**
+ * The programmatic identifier of the auxiliaryAuthenticationHeaderPolicy.
+ */
+export const auxiliaryAuthenticationHeaderPolicyName = "auxiliaryAuthenticationHeaderPolicy";
+const AUTHORIZATION_AUXILIARY_HEADER = "x-ms-authorization-auxiliary";
+
+/**
+ * Options to configure the auxiliaryAuthenticationHeaderPolicy
+ */
+export interface AuxiliaryAuthenticationHeaderPolicyOptions {
+  /**
+   * TokenCredential list used to get token from auxiliary tenants and
+   * one credential for each tenant the client may need to access
+   */
+  credentials?: TokenCredential[];
+  /**
+   * Scopes depend on the cloud your application runs in
+   */
+  scopes: string | string[];
+  /**
+   * A logger can be sent for debugging purposes.
+   */
+  logger?: AzureLogger;
+}
+
+type NullableString = string | null | undefined;
+
+async function sendAuthorizeRequest(options: AuthorizeRequestOptions): Promise<NullableString> {
+  const { scopes, getAccessToken, request } = options;
+  const getTokenOptions: GetTokenOptions = {
+    abortSignal: request.abortSignal,
+    tracingOptions: request.tracingOptions,
+  };
+
+  return (await getAccessToken(scopes, getTokenOptions))?.token;
```

Comment:
The return type of this is `Promise<string | undefined>` since the [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) will always return undefined if the field being accessed is undefined or null.

However, I think it might be simpler to default an empty token to the empty string since our filter for falsy values in the results will also take care of those:

```suggestion
  return (await getAccessToken(scopes, getTokenOptions))?.token ?? "";
```

This way we can remove the `NullableString` type entirely.